### PR TITLE
script.featherence.service [0.2.24]

### PR DESCRIPTION
**Pull request by finalmakerr**
Following: https://github.com/xbmc/repo-scripts/pull/107

### CODE REVIEW

**Comment by learningit** [comment]
What are the download routines doing?  Why through an external script?
There are some sleeps that are commented as being pretty long - up to 45 seconds. Why?

**Comment by finalmakerr** [comment]
Hey m8,
Thank you for your reply!

Could you be so kind and reference the file and module name you're referring?

File: script.featherence.service/resources/lib/shared/commondownloader.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            size = int(size)
+            req.add_header('Range',   'bytes=%d-' % size)
+
+        resp = urllib2.urlopen(req, timeout=10)
+        return resp
+    except:
+        return None
+
+
+def download(url, dest, title=None, referer=None, agent=None, cookie=None, silent=False):
```
Comment (learningit) [inline script.featherence.service/resources/lib/shared/commondownloader.py]: Here.

File: script.featherence.service/resources/lib/shared/commondownloader.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+            if hasattr(e, 'errno'):
+                errno = e.errno
+
+            if errno == 10035: # 'A non-blocking socket operation could not be completed immediately'
+                pass
+
+            if errno == 10054: #'An existing connection was forcibly closed by the remote host'
+                errors = 10 #force resume
+                sleep  = 30
```
Comment (learningit) [inline script.featherence.service/resources/lib/shared/commondownloader.py]: Here.

Comment (learningit) [inline script.featherence.service/resources/lib/shared/commondownloader.py]: Somewhere else (there's a lot of files here) I'm pretty sure I saw a m3u8 download routine or a multi-part downloader with lots of sleeps in it. First I would question the need for a downloading a content file and then the issues with so many long sleeps.

**Comment by phil65** [comment]
Sorry, but this script is such an unreviewable mess, I´m not really in favour of adding it to our repo...

**Comment by finalmakerr** [comment]
Hey Learningit,

Thank you for your reply!
The commondownloader.py is being used by Featherence Emu addon:
https://www.youtube.com/watch?v=3g_V95g2S2E

I removed it from Feathernece Service already!

Hey Phil,

Featherence Service is used in all of my addons, in order to see if it's interesting i suggest to try some of my addons first.

PS: I am doing my best if there are any suggestions, questions and will please share them with me :)

**Comment by finalmakerr** [comment]
Phil,
Could you be so kind to reply m8?

File: script.featherence.service/resources/lib/shared/shared_modules3.py [hunk 3]
```diff
@@ -970,7 +970,7 @@ def apimaster(x, title="", thumb="", desc="", fanart="", playlist=[], addonID=ad
 	
 	if url != "":
 		if 'A' in printpoint: title2 = ""
-		if not 'c' in printpoint:
+		if not 'c' in printpoint or 1 + 1 == 2:
```
Comment (phil65) [inline script.featherence.service/resources/lib/shared/shared_modules3.py]: really?

**Comment by phil65** [comment]
Sorry, as said in my opinion this script is too messy. I wont merge it, perhaps someone else wants.

**Comment by Tadly** [comment]
As apparent from what phil said, there's no nice way of putting it.
I had a browse through the source and there's just to much wrong (don't take that personal please).

I'm not sure how experienced with python you are but I suggest having a read through some PEP rules.

**Pull request closed without merging**